packages/datadog-instrumentations/src/langchain.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const { addHook } = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const tracingChannel = require('dc-polyfill').tracingChannel
+
+const invokeTracingChannel = tracingChannel('apm:langchain:invoke')
+
+function wrapLangChainPromise (fn, type, namespace = []) {
+  return function () {
+    if (!invokeTracingChannel.start.hasSubscribers) {
+      return fn.apply(this, arguments)
+    }
+
+    // Runnable interfaces have an `lc_namespace` property
+    const ns = this.lc_namespace || namespace
```

Comment:
`lc_namespace` may (in fact, wrt the test assertions, does) not be exactly the same as our Python integration, but we don't have access to something equivalent to `instance.__module__, instance.__class__.__name__` for setting this, and luckily there is `lc_namespace` variable. For cases where there isn't (ie Embeddings), I've opted to spelling out the namespace directly to mimic our python integration

Reply:
Understandable, as long as it's relatively clear to users where this span originated from (what langchain object/function is represented by this span) I'm happy with, cc @Kyle-Verhoog

Reply:
sounds good. will wait to see what Kyle thinks, but I also wanna call out that I'm not adding the function name to this, it's just something like `langchain_core.base.RunnableSequence`, or smth like that.

Reply:
Yup that's fine, as long as the original class/object is exposed that should be fine. Is there no way to access the name of the wrapped function?

Reply:
there is (can just do `func.name` in js), but I believe we already don't do this in the Python integration.

Reply:
yeah, from what i've seen, when `lc_namespace` is present, and we join it with the class name, it's a pretty good representation of the object/resource making the call (ie `langchain.llms.openai.OpenAI` or `langchain.llms.openai.ChatOpenAI`). otherwise, we can construct a representative resource (like we do for embeddings)

i think we're in agreement here, resolving this thread 😄